### ISSUE TITLE
Does Galaxy need multi-regionality?

### ISSUE THREAD

**Issue by chiniforooshan** [CloudVE/cloudbridge#54]
An AWS VPC is created in a single region and AWS supports adding one or more subnets in each availability zone. On the other hand, in GCP, a VPC is a global resource, spanning all regions. GCP supports adding subnets in each region. IMO, the simplest way to unify the API in CloudBridge is to restrict everything to a project-default region (and perhaps get rid of the region service/resource altogether). Does this look acceptable to you @nuwang, @afgane?

CC: @mbookman @baizhang

**Comment by afgane** [comment]
Nuwan and I had a chat about this just now and feel that a better solution would be to keep the interface as is; namely, require the `zone` to be supplied as a parameter when creating a `Subnet`. Then, internal to the GCE provider, the region to which the supplied zone belongs to can be extracted and used in the request. For example, OpenStack implementation adheres to the interface but simply [ignores the `zone` argument](https://github.com/gvlproject/cloudbridge/blob/master/cloudbridge/cloud/providers/openstack/services.py#L771).

Another point that came up while looking at this is use of `provider.default_zone` in several places in the GCE implementation (e.g., instance [reboot](https://github.com/gvlproject/cloudbridge/blob/gce/cloudbridge/cloud/providers/gce/resources.py#L834) method). Using `default_zone` in methods that don't create resources, including the `provider` should be removed. This is because if, for example, a zone was specified when creating an instance yet that zone is different from the `default_zone`, the query won't work as expected. Instead, the resource zone should be extracted from the `selfLink`. 
The only place this won't work right away is in the `InstanceTypeService`, [`instance_data` method](https://github.com/gvlproject/cloudbridge/blob/gce/cloudbridge/cloud/providers/gce/services.py#L282) because there is no local resource object. Instead, it looks like using [`aggregatedList` method](https://cloud.google.com/compute/docs/reference/beta/machineTypes/aggregatedList) instead of the `list` method would do the trick and not require a `zone` to be specified.

**Comment by afgane** [comment]
I just looked at the use of `default_zone` throughout the GCE implementation and I think the issue we identified in the previous comment has been corrected at this point. I feel the only potential issue with the use of that parameter may come up in the `Snapshot.create` method where the snapshot is being created in the `default_zone`. Instead, maybe we should be creating it in the `volume.zone_id` zone. And it's a big maybe (as in probably not) because the `volumes.list` method will return only the volumes from the current default region so in principle a given CloudBridge session should not even see the volumes outside the `default_zone`. 

Unless I'm missing something here, I move we close this issue.
